kitsune/sumo/static/sumo/js/wiki.js

```diff
@@ -627,99 +627,87 @@ import collapsibleAccordionInit from "sumo/js/protocol-details-init";
 
   function initRevisionList() {
     var $form = $('#revision-list form.filter');
+    
     if (!$form.length) {
-      return;
+        return;
     }
 
-    // This function grabs a fragment from the server and replaces
-    // the content of the div with it.
-    function updateRevisionList(query) {
-      if (query === undefined) {
-        query = $form.serialize();
-      }
-      if (query.charAt(0) !== '?') {
-        query = '?' + query;
-      }
-      var url = $form.attr('action') + query;
-
-      // scroll to the top.
-      var scrollPos = Math.min($(document).scrollTop(),
-        $('#revision-list').offset().top);
-      $(document).scrollTop(scrollPos);
-      history.replaceState({}, '', url);
-      $('#revisions-fragment').css('opacity', 0);
-
-      $.get(url + '&fragment=1', function (data) {
-        $('.loading').hide();
-        $('#revisions-fragment').html(data).css('opacity', 1);
-      });
+    // Retrieve and safely parse form data from localStorage
+    let formData = {};
+    const savedFormData = localStorage.getItem('revision-list-filter');
```

Comment:
Probably a better option for the scenario at hand is the `sessionStorage`. Using `sessionStorage` will ensure that we won't have any stale data and we won't also exceed the capacity.